staging/src/k8s.io/controller-manager/app/serve.go

```diff
@@ -55,7 +58,7 @@ func BuildHandlerChain(apiHandler http.Handler, authorizationInfo *apiserver.Aut
 }
 
 // NewBaseHandler takes in CompletedConfig and returns a handler.
-func NewBaseHandler(c *componentbaseconfig.DebuggingConfiguration, healthzHandler http.Handler) *mux.PathRecorderMux {
+func NewBaseHandler(c *componentbaseconfig.DebuggingConfiguration, healthzHandler http.Handler, componentName string) *mux.PathRecorderMux {
```

Comment:
This is a backward incompatible change to a published API function.

Maybe use something like a varargs argument where additional params can be provided, e.g:  `, ...HandlerOptions)` ?

You'll need to decide how to report component names of components that do not pass in a componentName.

Reply:
I just realize that our KEP wants to enable statusz in *kube*-controller-manager. So remove the *cloud*-controller-manager change.

Reply:
That sounds good.  The `genericcontrollermanager.NewBaseHandler` call is also being made from `cmd/`kube-controller-manager` so my above comment still applies.